torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -503,7 +503,7 @@ class TestDistBackend(MultiProcessTestCase):
     @classmethod
     def setUpClass(cls):
         os.environ["MASTER_ADDR"] = str(MASTER_ADDR)
-        os.environ["MASTER_PORT"] = str(MASTER_PORT)
+        # Not setting MASTER_PORT and get a random free port
```

Comment:
is there a reason we need to change to the random port? is this because we run tests in parallel and two subprocess trying to access the same port?

Reply:
Yeah, that what I suspect. When this port was fixed, `TestDistBackendWithSpawn.test_DistributedDataParallel_non_default_stream` failed with the following error:

```
ERROR:torch.testing._internal.common_distributed:Caught exception: 
Traceback (most recent call last):
  File "/opt/conda/lib/python3.10/site-packages/torch/testing/_internal/common_distributed.py", line 622, in run_test
    getattr(self, test_name)()
  File "/opt/conda/lib/python3.10/site-packages/torch/testing/_internal/common_distributed.py", line 503, in wrapper
    fn()
  File "/opt/conda/lib/python3.10/site-packages/torch/testing/_internal/common_distributed.py", line 145, in wrapper
    return func(*args, **kwargs)
  File "/opt/conda/lib/python3.10/site-packages/torch/testing/_internal/distributed/distributed_test.py", line 4139, in test_DistributedDataParallel_non_default_stream
    self.assertEqual(
  File "/opt/conda/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 2402, in assertEqual
    assert_equal(
  File "/opt/conda/lib/python3.10/site-packages/torch/testing/_comparison.py", line 1093, in assert_equal
    raise error_metas[0].to_error(msg)
AssertionError: Scalars are not close!

Absolute difference: 0.25 (up to 1e-05 allowed)
Relative difference: 0.5 (up to 1.3e-06 allowed) : Expected gradient of 0.5 but got tensor([[0.2500]], device='cuda:0') on rank 0
 exiting process 0 with exit code: 10
ERROR:torch.testing._internal.common_distributed:Caught exception: 
Traceback (most recent call last):
  File "/opt/conda/lib/python3.10/site-packages/torch/testing/_internal/common_distributed.py", line 622, in run_test
    getattr(self, test_name)()
  File "/opt/conda/lib/python3.10/site-packages/torch/testing/_internal/common_distributed.py", line 503, in wrapper
    fn()
  File "/opt/conda/lib/python3.10/site-packages/torch/testing/_internal/common_distributed.py", line 145, in wrapper
    return func(*args, **kwargs)
  File "/opt/conda/lib/python3.10/site-packages/torch/testing/_internal/distributed/distributed_test.py", line 4139, in test_DistributedDataParallel_non_default_stream
    self.assertEqual(
  File "/opt/conda/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 2402, in assertEqual
    assert_equal(
  File "/opt/conda/lib/python3.10/site-packages/torch/testing/_comparison.py", line 1093, in assert_equal
    raise error_metas[0].to_error(msg)
AssertionError: Scalars are not close!

Absolute difference: 0.25 (up to 1e-05 allowed)
Relative difference: 0.5 (up to 1.3e-06 allowed) : Expected gradient of 0.5 but got tensor([[0.2500]], device='cuda:1') on rank 1
 exiting process 1 with exit code: 10
Process 0 terminated with exit code 10, terminating remaining processes.
```

Here is an example https://github.com/pytorch/pytorch/runs/8009341997.  I have a follow-up action item to understand why it happens, so really appreciate if you can point me to the right direction :)

Reply:
Oh, it turns out that this issue come from a completely different direction. This test is flaky https://github.com/pytorch/pytorch/issues/76428 and has been like that for a while https://www.torch-ci.com/failure/test_DistributedDataParallel_non_default_stream%2C%20TestDistBackendWithSpawn.  Now it's back hmm